### ISSUE TITLE
allow specifying a single sketch to use for intersections in `preston diff`

### ISSUE THREAD

**Issue by mielliott** [bio-guoda/preston#129]
Current behavior of `preston diff` is to do an "all-to-all" comparison, where preston computes and reports the intersections of each sketch with every other sketch. This requires at least `(n^2)/2` computations. I'd like to be able to do a "one-to-all" comparison, to compute the intersection of every sketch with one specific sketch, which should require only `n` computations. Maybe we could allow users to specify one sketch to be compared with all sketches found in the input, e.g.
```
$ preston ls | preston grep | preston bloom | preston diff hash://sha256/abc...
```
so that every hash streamed into `preston diff` will only be intersected with the bloom filter identified by `hash://sha256/abc...`. If no hash is specified, then default to the current "all-to-all" behavior.

## Use case

I'd like to be able to use bloom filters/theta sketches to quickly find all datasets that contain at least one of a large set of terms that I'm looking for. For example, if I have a list of 1000 species names, I would
1. compute a bloom filter for that set of names; let's say the filter is saved as `hash://sha256/def...`
1. use `preston ls | preston grep [some regex] | preston bloom | preston diff hash://sha256/def` to find which datasets contain at least one of those 1000 species names

**Comment by jhpoelen** [comment]
@mielliott neat! If I understand your correctly, you are proposing to turn preston diff into a way to query any collection of arbitrary datasets. 

Is that what you had in mind?

**Comment by mielliott** [comment]
@jhpoelen exactly! For big datasets like deeplinker.bio, computing n^2 intersections might not complete in my lifetime (although I would encourage future generations to pick up our work if needed), but for some applications, having an O(n) query method could be used to narrow the search space enough to reasonably compute m^2 (where m << n) all-to-all intersections.

**Comment by jhpoelen** [comment]
@mielliott Although I like the idea of compute job that spans multiple generations, I can see now a O(n) query method is a little more pragmatic. Do you have an example to work with?

**Comment by mielliott** [comment]
For sure: @JimmyFromRobotics and I would like to try to find biodiversity datasets representing physical collections housed in the United States in order to find datasets that iDigBio is missing (maybe GBiF/etc. knows about them, but iDigBio doesn't). One approach might be to
1. build our own list of key terms and phrases whose presence in a dataset might indicate that the physical collection is housed in the US,
1. construct a bloom filter via something like `preston update file:///ourterms.txt | preston grep .+ | preston bloom`,
1. use `preston ls | preston grep [pattern] | preston bloom [hash for our filter] --remote https;//deeplinker.bio/` to search deeplinker.bio for datasets whose bloom filters intersect with our key terms filter, i.e., they likely contain one of the terms we're looking for.

The `[pattern]` to grep with is the real magic sauce we need to figure out, and maybe `grep` isn't the best tool for this, but I'm sure the results/process will be interesting in any case.
